aten/src/ATen/native/cuda/AveragePool2d.cu

```diff
@@ -183,8 +183,8 @@ __global__ void avg_pool2d_backward_out_cuda_frame(const int nthreads, const sca
   }
 }
 
-template <typename scalar_t, typename accscalar_t>
-__global__ void avg_pool2d_backward_out_cuda_frame_nhwc(const int nthreads,
+template <typename scalar_t, typename accscalar_t, typename index_t>
+__global__ void avg_pool2d_backward_out_cuda_frame_nhwc(const index_t nthreads,
     const scalar_t* const top_diff,
     const int64_t channels, const int64_t height,
     const int64_t width, const int pooled_height, const int pooled_width,
```

Comment:
noob question that is maybe unrelated: why are `pooled_height` and `pooled_width` `const int` here in `avg_pool2d_backward_out_cuda_frame_nhwc` but `const int64_t` above in  `avg_pool2d_backward_out_cuda_frame`?

Reply:
It doesn't look obvious to me as I can't immediately see where its use/definition diverges between the channels-first and channels-last cases. Maybe @xwang233 can comment as he authored the channels-last version (https://github.com/pytorch/pytorch/pull/35855) and the last round of 64-bit indexing fixes.

Reply:
btw I think my question is unrelated to the initial issue, did not mean to block this PR from merging. Could you perhaps file an issue about the typing difference between channels-last and non-channels-last for avg_pool though so this doesn't get lost please?